### PR TITLE
Reduce access token refreshing if it is still fresh

### PR DESCRIPTION
**Pull request by licitdev**
## Scope

What's changed:

- Reimplemented the skipping of access token refresh logic while it is still fresh

## Potential Risks / Drawbacks

- Should not have any as the access token generally should remain valid up till its expiry

## Review Notes / Questions

- This logic was removed in #21938

https://github.com/directus/directus/blob/77d73398f1be1501ee9196b9fd384666a0a9712b/app/src/auth.ts#L99-L105

### CODE REVIEW

File: app/src/auth.ts [hunk 1]
```diff
@@ -94,6 +94,9 @@ export async function refresh({ navigate }: LogoutOptions = { navigate: true }):
 	// Allow refresh during initial page load, skip if not logged in
 	if (!firstRefresh && !appStore.authenticated) return;
 
+	// Skip access token refreshing if it is still fresh
+	if (appStore.accessTokenExpiry && Date.now() < appStore.accessTokenExpiry - SDK_AUTH_REFRESH_BEFORE_EXPIRES) return;
```
Comment (br41nslug) [inline app/src/auth.ts:98]:
```suggestion
	if (appStore.accessTokenExpiry && Date.now() < appStore.accessTokenExpiry - SDK_AUTH_REFRESH_BEFORE_EXPIRES) {
		try {
			await sdk.request(readMe());
		} catch {
			await logout({ navigate, reason: LogoutReason.SESSION_EXPIRED });
		}

		return;
	}
```

Comment (licitdev) [reply app/src/auth.ts:98]: 👍 Shall we fetch the `id` field with `sdk.request(readMe({ fields: ['id'] }))`, reducing the response body size?

Comment (br41nslug) [reply app/src/auth.ts:98]: Good suggestion, should we keep that "SESSION EXPIRED" reason?

Comment (licitdev) [reply app/src/auth.ts]: Yes, I believe so, and it's consistent with the existing logic.
Changes were pushed in https://github.com/directus/directus/pull/22464/commits/dfa4a078ea9e3d33466e89074087f7e39bf3d2de where the session validation is added into the existing try-catch-finally clause.

File: .changeset/bright-trains-greet.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@directus/app': patch
+---
+
+Reduced access token refreshing if it is still fresh
```
Comment (hanneskuettner) [inline .changeset/bright-trains-greet.md:5]: Maybe?

```suggestion
Reduced the number of session token refreshes if the token is still fresh
```

**Review by hanneskuettner — APPROVED** [review]
Nice fix! :guitar:

I've verified with a short session cookie duration that both the SDK triggered refresh and the tab active refresh work as expected, and otherwise the `/users/me` endpoint is called.

File: .changeset/bright-trains-greet.md (hunk 2 above)
Comment (licitdev) [reply .changeset/bright-trains-greet.md:5]: That works! Thanks! 👍

**Review by paescuj — APPROVED** [review]

**Pull request merged**